src/Components/ArticleById.jsx

```diff
@@ -1,7 +1,58 @@
-import { useState } from "react";
+import { useEffect, useState } from "react";
+import { getArticleById } from "../api";
+import { useParams } from "react-router-dom";
+import moment from "moment";
 
 function ArticleById() {
-   const [singleArticle, setSingleArticle] = useState("");
+   const { articleId } = useParams();
+   const [singleArticle, setSingleArticle] = useState([]);
+
+   useEffect(() => {
+      getArticleById(articleId).then((article) => {
+         setSingleArticle(article);
+      });
+   }, [articleId]);
+
+   if (!singleArticle) {
```

Comment:
This would render initially, when trying to load the article which isn't ideal for the user

Reply:
It would be better to have some loading state and indication